magicbot/magic_tunable.py

```diff
@@ -123,34 +125,93 @@ def setup_tunables(component, cname, prefix='components'):
         prop._ntattr = ntattr
 
 
-# TODO
-#def autosend(f=None):
-#    '''
-#        Decorator used to send variables to DS::
-#        
-#            
-#        
-#            class MyComponent:
-#            
-#                @autosend
-#                def my_sensor(self):
-#                    return self.limit_switch.get()
-#                    
-#                ...
-#                    
-#            class MyRobot:
-#            
-#                mine = MyComponent 
-#                
-#        This will cause the output of this function to be sent
-#        to ``/components/mine/my_sensor``
-#    '''
-#    
-#    
-#    def _get(self):
-#        return self._Magicbot__autosend.get(f)
-#    
-#    prop = _AutosendProperty(fget=_get)
-#    prop.f = f
-#    
-#    return prop
+def feedback(f=None, *, key: str = None):
```

Comment:
I imagine you know what you're doing, but what is the f=None?

Reply:
`f` will be the function object being decorated. The default value of `None` and the following if statement allows the user to use the decorator without requiring the use of extra parentheses (see the given example).

This works similarly to our other decorators.

Reply:
ah